src/constraints.jl

```diff
@@ -152,14 +152,32 @@ function MOI.get(m::LinQuadOptimizer, ::Type{MOI.ConstraintIndex{F,S}}, name::St
     m.constraint_names_rev[name]::MOI.ConstraintIndex{F,S}
 end
 
+
+"""
+    CSRMatrix{T}
+
+Matrix given in compressed sparse row (CSR) format.
+
+It contains three vectors:
+ - `column_indices` is a vector of column numbers;
+ - `data` is a vector of corresponding nonzero values; and
+ - `row_pointers` is a vector pointing to the start of each row in
+    `column_indices` and `data`.
+
+The length of `row_pointers` is the number of rows in the matrix.
+"""
+struct CSRMatrix{T}
+    column_indices::Vector{Int}
+    data::Vector{T}
+    row_pointers::Vector{Int}
+    function CSRMatrix{T}(column_indices, data, row_pointers) where T
+        @assert length(column_indices) == length(data)
```

Comment:
Where does additional validation occur? Is it on the solver to crash? Add a note that consumers of `CSRMatrix` shouldn't assume that the matrix is well formed if that's the case.

Reply:
It doesn't even look like SparseMatrixCSC does validation:
https://github.com/JuliaLang/julia/blob/f7ed09baa0cdafff1141c796cbdfb9c6d1e6c406/stdlib/SparseArrays/src/sparsematrix.jl#L21-L29

Reply:
Just document that consumers of the structure shouldn't assume the data has been validated.